cli/src/cli_util.rs

```diff
@@ -1006,6 +1006,14 @@ impl WorkspaceCommandHelper {
         ui: &mut Ui,
     ) -> Result<Commit, CommandError> {
         let revset_expression = self.parse_revset(revision_str, Some(ui))?;
+        let branch: Option<String> =
+            if let RevsetExpression::CommitRef(RevsetCommitRef::Symbol(ref branch_name)) =
+                *revset_expression
+            {
+                Some(branch_name.clone())
+            } else {
+                None
+            };
```

Comment:
We shouldn't have to do this in the common case when the revset resolved to a single commit, so can you move it into the block where it's used? Then I don't think the `branch` variable is even needed (and you can drop the `.clone()` call too).

Reply:
I found it strange too, but I did this because `revset_expression` is moved in the next line. In retrospect I could just do this to avoid `revset_expression` from being moved thought (I will do this)
```diff
- let revset = self.evaluate_revset(revset_expression))?;
+ let revset = self.evaluate_revset(revset_expression.clone())?;
```